src/internal/git/push.go

```diff
@@ -37,55 +63,68 @@ func PushAllDirectories(localPath string) error {
 	for _, path := range paths {
 		basename := filepath.Base(path)
 		spinner.Updatef("Pushing git repo %s", basename)
-		if err := push(path, tunnelUrl, spinner); err != nil {
+
+		repo, err := prepRepoForPush(path, gitServerURL, gitServerInfo.GitPushUsername)
+		if err != nil {
+			message.Warnf("error when preping the repo for push.. %v", err)
+			return err
+		}
+
+		if err := push(repo, path, spinner); err != nil {
 			spinner.Warnf("Unable to push the git repo %s", basename)
 			return err
 		}
 
 		// Add the read-only user to this repo
-		repoPathSplit := strings.Split(path, "/")
-		repoNameWithGitTag := repoPathSplit[len(repoPathSplit)-1]
-		repoName := strings.Split(repoNameWithGitTag, ".git")[0]
-		err = addReadOnlyUserToRepo(tunnelUrl, repoName)
-		if err != nil {
-			message.Warnf("Unable to add the read-only user to the repo: %s\n", repoName)
-			return err
+		if gitServerInfo.InternalServer {
```

Comment:
We should probably think about this some more--we intentionally kept the read only user creation in the git server component so it wasn't tightly-coupled/hard-coded.  

It may require a different PR, but I think I've found a better way that exists now in Gitea to handle read only users:  you can now create repos per-org on push and then assign a user to the org once--or just set the rules to "must be logged in" to gitea and even that isn't required.    https://docs.gitea.io/en-us/config-cheat-sheet/#repository-repository, `ENABLE_PUSH_CREATE_ORG`. 


<img width="1283" alt="Screen Shot 2022-08-01 at 1 42 19 PM" src="https://user-images.githubusercontent.com/882485/182221322-6b945ff9-7047-4a7a-b9a3-dd2058f3ac96.png">
<img width="1290" alt="Screen Shot 2022-08-01 at 1 42 11 PM" src="https://user-images.githubusercontent.com/882485/182221332-5960c396-6355-4b10-b1c6-b8e46795c7bd.png">
<img width="1292" alt="Screen Shot 2022-08-01 at 1 41 34 PM" src="https://user-images.githubusercontent.com/882485/182221337-20d0f49c-1974-4f55-9a30-e8be3355f6ad.png">
<img width="1289" alt="Screen Shot 2022-08-01 at 1 40 53 PM" src="https://user-images.githubusercontent.com/882485/182221339-ebeafb34-9cb2-4e53-ac4f-acb568f529f5.png">
<img width="929" alt="Screen Shot 2022-08-01 at 1 39 55 PM" src="https://user-images.githubusercontent.com/882485/182221340-133c3f6f-710e-4081-862d-9437d9795337.png">
<img width="839" alt="Screen Shot 2022-08-01 at 1 38 55 PM" src="https://user-images.githubusercontent.com/882485/182221344-601e091a-a3cf-4185-8f5d-9a0561a66c6a.png">